<launch>
    <arg name="cmd_src" default="teleop" />
    <arg name="robot" default="nusim"/>

    <!--  -->
    <group if="$(eval '\'$(var robot)\' == \'nusim\'')">
        <!-- Include the start_robot launch file w/ nusim & teleop control -->
        <include file="$(find-pkg-share nuturtle_control)/launch/start_robot.launch.xml">
            <arg name="cmd_src" value="$(var cmd_src)"/>
        </include>

        <!-- Run the slam node with the necessary parameters and remapping -->
        <node pkg="nuslam" exec="nuslam">
            <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
            <param name="body_id" value="green/base_footprint"/>
            <param name="wheel_left" value="wheel_left_joint"/>
            <param name="wheel_right" value="wheel_right_joint"/>
            <remap from="joint_states" to="red/joint_states" />
            <remap from="nav_msgs/path" to="green/nav_msgs/path" />
            <remap from="fake_sensor" to="nusim/fake_sensor" />
        </node>

        <!-- Load a green turtlebot -->
        <group>
            <push-ros-namespace namespace="green"/>
            <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py" >
                <arg name="color" value="green"/>
                <arg name="use_rviz" value="false"/>
                <arg name="use_jsp" value="true"/>
            </include>
        </group>
    </group>

</launch>
